File: content/development/golang-first-impression/data_structure.cs
using System;
// for List, Dictionary
using System.Collections.Generic;

public class Simple {
    public Simple(int a)
    {
        this.a = a;
    }
    public int a;
}

public class Program
{
    public static void Main()
    {
        // list
        List<Simple> dummyList = new List<Simple>()
            {
                new Simple(1),
            };
        dummyList.Add(new Simple(2));
        foreach(Simple s in dummyList) {
            Console.WriteLine("{0} = {1}", s, s.a);
        }

        // dictionary
        Dictionary<string, Simple> dummyMap = new Dictionary<string, Simple>()
            {
                { "foo", new Simple(1) },
            };
        dummyMap["bar"] = new Simple(2);
        foreach(KeyValuePair<string, Simple> kv in dummyMap) {
            Console.WriteLine("{0} = {1}", kv.Key, kv.Value);
        }
    }
}

File: content/posts/development/golang-first-impression/maya/data_structure.go
package main

import "fmt"

type Simple struct {
	a int
}

func main() {
	// slice
	dummySlice := []Simple{{1}}
	dummySlice = append(dummySlice, Simple{2})
	for i, v := range dummySlice {
		fmt.Println(i, v)
	}

	// map
	dummyMap := map[string]Simple{
		"foo": {1},
	}
	dummyMap["bar"] = Simple{2}
	for k, v := range dummyMap {
		fmt.Println(k, v)
	}
}
